runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -3664,6 +3664,151 @@ TR::Register *J9::X86::TreeEvaluator::longBitCount(TR::Node *node, TR::CodeGener
+      {
+      // touch the memory in case this is a NULL pointer and we need to throw
+      auto instr = generateMemImmInstruction(TEST1MemImm1, node, generateX86MemoryReference(ObjReg, TR::Compiler->om.offsetOfObjectVftField(), cg), 0, cg);
+      cg->setImplicitExceptionPoint(instr);
+      instr->setNeedsGCMap(0xFF00FFFF);
+      instr->setNode(cg->comp()->findNullChkInfo(node));
+      }
+   generateLabelInstruction(JMP4, node, outlinedCallLabel, cg);
+   
+   // Object not array, inline checks
+   generateLabelInstruction(LABEL, node, notArrayLabel, cg);
+   
+   // Null test
+   if (node->getOpCodeValue() != TR::checkcastAndNULLCHK)
+      {
+      generateRegRegInstruction(TESTRegReg(), node, ObjReg, ObjReg, cg);
+      generateLabelInstruction(JE4, node, fallThruLabel, cg);
+      }
+   
+   // load object class in temp3
+   generateLoadJ9Class(node, temp3Reg, ObjReg, cg);
+   
+   // cast class is interface
+   generateMemImmInstruction(TEST4MemImm4, node,
+       generateX86MemoryReference(temp2Reg, offsetof(J9ROMClass, modifiers), cg), J9AccInterface, cg);
+   generateLabelInstruction(JE4, node, isClassLabel, cg);
+   
+   // Obtain I-Table
+   generateRegMemInstruction(LRegMem(), node, temp1Reg, generateX86MemoryReference(temp3Reg, offsetof(J9Class, iTable), cg), cg);
+   // Loop through I-Table
+   generateLabelInstruction(LABEL, node, iTableLoopLabel, cg);
+   generateRegRegInstruction(TESTRegReg(), node, temp1Reg, temp1Reg, cg);
+   generateLabelInstruction(JE4, node, throwLabel, cg);
+   auto interfaceMR = generateX86MemoryReference(temp1Reg, offsetof(J9ITable, interfaceClass), cg);
+   generateMemRegInstruction(CMPMemReg(), node, interfaceMR, castClassReg, cg);
+   generateRegMemInstruction(LRegMem(), node, temp1Reg, generateX86MemoryReference(temp1Reg, offsetof(J9ITable, next), cg), cg);
+   generateLabelInstruction(JNE4, node, iTableLoopLabel, cg);
+   
+   // Found from I-Table
+   generateLabelInstruction(JMP4, node, fallThruLabel, cg);
+   
+   // cast class is non-interface class
+   generateLabelInstruction(LABEL, node, isClassLabel, cg);
+   // equality test
+   generateRegRegInstruction(CMPRegReg(use64BitClasses), node, temp3Reg, castClassReg, cg);
+   generateLabelInstruction(JE4, node, fallThruLabel, cg);
+   
+   // class not equal,
+   // if cast class depth >= min depth, compare with object class depth
+   // store cast class depth in temp2 for later use
+   // class depth mask must be low 16 bits to safely load without the mask.
+   static_assert(J9AccClassDepthMask == 0xffff, "J9_JAVA_CLASS_DEPTH_MASK must be 0xffff");
+   generateRegMemInstruction(TR::Compiler->target.is64Bit()? MOVZXReg8Mem2 : MOVZXReg4Mem2, node,
+            temp2Reg, generateX86MemoryReference(castClassReg, offsetof(J9Class, classDepthAndFlags), cg), cg);
+   generateRegImmInstruction(CMP2RegImm2, node, temp2Reg, ((int16_t)(cg->comp()->getOptions()->_minimumSuperclassArraySize)), cg);
```

Comment:
Checking if `cast class depth >= min depth` is unnecessary, as the checking if `cast class depth >= obj class depth` is sufficient to conclude if testing superclass is required.

Reply:
Addressed. Removed the extraneous min depth check.